outboundSender.go

```diff
@@ -105,70 +101,52 @@ type OutboundSender interface {
 
 // CaduceusOutboundSender is the outbound sender object.
 type CaduceusOutboundSender struct {
```

Comment:
updated `CaduceusOutboundSender` to simplify and prep for the fx integration by introducing `OutboundSenderMetrics`